lib/incoming_mail.rb

```diff
@@ -0,0 +1,110 @@
+module IncomingMail
+  # mail_types
+  UNKNOWN = 0
+  REQUEST = 1
+
+  class Handler
+    # parses a Mandrill callback object and Does The Right Thing
+
+    def initialize(params = {})
+      params.each do |key, value|
+        instance_variable_set("@#{key}", value)
+      end
+    end
+
+    def handle(payload)
+      if payload.is_a?(Array) and payload[0]['event'] and payload[0]['event'] == 'inbound'
+        create_response(payload[0])
+      elsif payload.is_a?(Mandrill::WebHook::EventDecorator)
```

Comment:
interested in why there are 2 different possibilities here - perhaps a test for each case would clarify?